src/main/java/com/lunark/lunark/reservations/service/ReservationService.java

```diff
@@ -0,0 +1,83 @@
+package com.lunark.lunark.reservations.service;
+
+import com.lunark.lunark.auth.model.Account;
+import com.lunark.lunark.auth.repository.IAccountRepository;
+import com.lunark.lunark.properties.model.Property;
+import com.lunark.lunark.properties.model.PropertyAvailabilityEntry;
+import com.lunark.lunark.properties.repostiory.IPropertyRepository;
+import com.lunark.lunark.reservations.dto.ReservationRequestDto;
+import com.lunark.lunark.reservations.model.Reservation;
+import com.lunark.lunark.reservations.model.ReservationStatus;
+import com.lunark.lunark.reservations.repository.IReservationRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Service;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.security.Principal;
+import java.time.LocalDate;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class ReservationService implements IReservationService {
+    private final IReservationRepository reservationRepository;
+    private final IPropertyRepository propertyRepository;
+    private final IAccountRepository accountRepository;
+
+    @Autowired
+    public ReservationService(IReservationRepository reservationRepository, IPropertyRepository propertyRepository, IAccountRepository accountRepository) {
+        this.reservationRepository = reservationRepository;
+        this.propertyRepository = propertyRepository;
+        this.accountRepository = accountRepository;
+    }
+
+    @Override
+    public Reservation create(ReservationRequestDto reservationDto, String username) {
+        Optional<Property> property = propertyRepository.findById(reservationDto.getPropertyId());
+        if (property.isEmpty()) {
+            throw new ResponseStatusException(HttpStatus.NOT_FOUND, "property not found");
+        }
+
+        if (!property.get().isAvailable(reservationDto.getStartDate(), reservationDto.getEndDate())) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "property not available for given date range");
+        }
+
+        if (property.get().getMinGuests() > reservationDto.getNumberOfGuests() || property.get().getMaxGuests() < reservationDto.getNumberOfGuests()) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "number of guests does not match the defined guest range of property");
+        }
+
+        Optional<Account> guest = accountRepository.findByEmail(username);
+        if (guest.isEmpty()) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "user does not exist");
+        }
+
+        double totalPrice = calculatePrice(
+                reservationDto.getStartDate(),
+                reservationDto.getEndDate(),
+                property.get().getAvailabilityEntries().stream().toList()
```

Comment:
I suggest shortening this line by removing the .stream.toList() and changing the type of this parameter to Collection. `Collection`s support the same stream operation as `List`s. 
```suggestion
                property.get().getAvailabilityEntries()
```